### PR TITLE
sanitize class paths

### PR DESCRIPTION
**Pull request by vladak**
This change should fix the "zip slip" issue in plugin framework reported by Sonar. Normally the plugin archives are considered trusted however a bit of sanitization does not hurt.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java [hunk 1]
```diff
@@ -313,9 +319,24 @@ private String getClassName(File f) {
...
+    @Nullable
     private String getClassName(JarEntry f) {
+        final String filePath = f.getName();
+        if (!filePath.endsWith(CLASS_SUFFIX)) {
+            return null;
+        }
+
+        File file = new File(pluginDirectory.getAbsolutePath(), filePath);
+        try {
+            if (!file.getCanonicalPath().startsWith(pluginDirectory.getCanonicalPath())) {
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java:331]: Shouldn't the `File.separator` be appended?
```java
if (!file.getCanonicalPath().startsWith(pluginDirectory.getCanonicalPath() + File.separator)) {
```
At least the example on https://snyk.io/research/zip-slip-vulnerability has it.

File: opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java [hunk 2]
```diff
@@ -313,9 +319,24 @@ private String getClassName(File f) {
...
+                return null;
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java:332]: Maybe add a warning log?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java [hunk 3]
```diff
@@ -313,9 +319,24 @@ private String getClassName(File f) {
...
+            }
+        } catch (IOException e) {
+            return null;
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java:335]: I think this shouldn't happen often, maybe add a log?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java (hunk 1 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java:331]: Good point.

**Pull request merged**